### DISCUSSION TITLE
Server Functions with Axum Workspace

### DISCUSSION THREAD

**Discussion by DreamTexX** [leptos-rs/leptos#1611] [Q&A]
Is there a suggested location for the server functions when utilizing the [axum workspace template](https://github.com/leptos-rs/start-axum-workspace)? Should they be put in "app", "server", or a new package made for them?

**Comment by gbj** [comment]
I think `app` would be the typical answer here, since you need to be able to call them from components in your `app`, and `server` depends on `app` but not the other way around.

**Reply by paulvt** [reply]
I was wondering the same. I was trying to do the `<ActionForm/>` [part of the tutorial](https://leptos-rs.github.io/leptos/progressive_enhancement/action_form.html) with this template, updated with the 0.5.0 changes. However when I add this to `app/src/lib.rs`:
```rust
#[server]
pub async fn add_todo(title: String) -> Result<(), ServerFnError> {
    todo!()
}
```
the build fails:
```
error[E0463]: can't find crate for `serde`
 --> app/src/lib.rs:8:1
  |
8 | #[server]
  | ^^^^^^^^^ can't find crate
  |
  = note: this error originates in the derive macro `::leptos::server_fn::serde::Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)
```
and then the somewhat misleading error:
```
error: open tag has no corresponding close tag
  --> app/src/lib.rs:59:9
   |
59 |         <ActionForm action=add_todo>
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: treating all inputs after open tag as it content
```

Do I need to include more? If so, should I ask it to be added to the workspace template?

**Reply by gbj** [reply]
`cargo add serde --features=derive` on the first one. Not sure what the second one is.

**Reply by paulvt** [reply]
Alright, I already suspected that. Given that this template is going to be used because you want server functions, I'd say it should be in the workspace template by default :).
I will check if it makes the second error disappear.

**Reply by paulvt** [reply]
Verified, the error disappeared. I have opened [an issue](https://github.com/leptos-rs/start-axum-workspace/issues/6) for it to be included.

Also, the corresponding close tag was written as `/ActionForm>` which was obviously wrong and a typo by me.
